### PR TITLE
feat(new-trace): Updated logic for routing upon event id click from discover

### PR DESCRIPTION
**Pull request by Abdkhan14**
We should route to issue details for events that are not of type `transaction` or are not associated to a `trace` id

### CODE REVIEW

File: static/app/views/discover/table/tableView.tsx [hunk 1]
```diff
@@ -190,7 +190,10 @@ function TableView(props: TableViewProps) {
 
       let target;
 
-      if (dataRow.trace === null || dataRow.trace === undefined) {
+      if (
+        !dataRow.trace ||
+        ['error', 'default'].includes(dataRow['event.type'].toString())
```
Comment (Zylphrex) [inline static/app/views/discover/table/tableView.tsx:195]: I think this should be `dataRow['event.type'].toString() != 'transaction'` as we have other error event types such as csp and a few more that I don't recall.

File: static/app/views/discover/table/tableView.tsx [hunk 2]
```diff
@@ -321,7 +324,10 @@ function TableView(props: TableViewProps) {
     if (columnKey === 'id') {
       let target;
 
-      if (dataRow.trace === null || dataRow.trace === undefined) {
+      if (
+        !dataRow.trace ||
+        ['error', 'default'].includes(dataRow['event.type'].toString())
```
Comment (Zylphrex) [inline static/app/views/discover/table/tableView.tsx:329]: Same here

**Review by Zylphrex — COMMENTED** [review]
Just double checking but `event.type` is always defined right?

File: static/app/views/discover/table/tableView.tsx (hunk 1 above)
Comment (Abdkhan14) [reply static/app/views/discover/table/tableView.tsx:195]: Ahh I se

File: static/app/views/discover/table/tableView.tsx [hunk 3]
```diff
@@ -190,13 +190,7 @@ function TableView(props: TableViewProps) {
...
       let target;
 
-      if (dataRow.trace === null || dataRow.trace === undefined) {
-        if (dataRow['event.type'] === 'transaction') {
-          throw new Error(
-            'Transaction event should always have a trace associated with it.'
-          );
-        }
-
+      if (dataRow['event.type'] !== 'transaction') {
```
Comment (Zylphrex) [inline static/app/views/discover/table/tableView.tsx:193]: Nit: This is more of a stylistic thing but can we invert this condition

```typescript
if (dataRow['event.type'] === 'transaction) {
  // the stuff current in side else block
} else {
  // the stuff currently in the if block
}
```

Makes this easier to reason about at a glance with fewer negations.

File: static/app/views/discover/table/tableView.tsx [hunk 4]
```diff
@@ -321,13 +321,7 @@ function TableView(props: TableViewProps) {
...
```
Comment (Zylphrex) [inline static/app/views/discover/table/tableView.tsx:324]: Same thing here.

**Review by Zylphrex — APPROVED** [review]

**Pull request merged**